docs/backend-api/api-methods.md

```diff
@@ -193,19 +197,262 @@ You will need to generate new token after the first one expires.
+
+* **Request Body**
+
+  * **Content**
+    ```
+    {
+      "name": {
+          "en": "22\" Sencor SLE 22F46DM4 HELLO KITTY",
+          "cs": "22\" Sencor SLE 22F46DM4 HELLO KITTY"
+      },
+      "hidden": false,
+      "sellingDenied": false,
+      "sellingFrom": "2000-01-16T00:00:00+00:00",
+      "sellingTo": "2100-01-16T00:00:00+00:00",
+      "catnum": "9177759",
+      "ean": "8845781245930",
+      "partno": "SLE 22F46DM4",
+      "shortDescription": {
+          "1": "Television LED, 55 cm diagonal ...",
+          "2": "Sencor SLE 22F46DM4 Hello Kitty je ..."
+      },
+      "longDescription": {
+          "1": "Television LED, 55 cm diagonal ...",
+          "2": "<p><strong>Sencor SLE 22F46DM4 ..."
+      }
+    }
+    ```
+
+* **Success Response**
+
+  * **Code** `201`  
+    **Headers**
+    `Location: http://127.0.0.1:8000/api/v1/products/7a34a8cf-8f3b-405d-9662-f20d983cfc8a`
+
+* **Error Response**
+
+  * **Code** `401 UNAUTHORIZED`
+
+  OR
+
+  * **Code** `400 Bad Request`  
+    **Content**
+    ```
+    {
+        "message": "Provided data did not pass validation",
+        "errors": [
+            "[sellingTo]" => "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
+            "[longDescription][1]" => "The value 345 is not a valid string.",
+        ]
+    }
+    ```
+
+  OR
+
+  * **Code** `422 Unprocessable Entity`  
+    **Content**
+    ```
+    {
+        "message": "Product with 74da76db-61ab-43fb-926e-43ba326325fc UUID already exists"
+    }
+    ```
+
+**Update Product**
+----
+
+* **URL**
+
+  `/api/v1/products/{uuid}`
+
+* **Method**
+
+  `PATCH`
+
+*  **URL Params**
+
+   **Required**
+
+   `uuid=[string]`
+
+* **Request Body**
+
+  * **Content**
+    ```
+    {
+      "name": {
+          "en": "22\" Sencor SLE 22F46DM4 HELLO KITTY",
+          "cs": "22\" Sencor SLE 22F46DM4 HELLO KITTY"
+      },
+      "hidden": false,
+      "shortDescription": {
+          "1": "Television LED, 55 cm diagonal ..."
+      }
+    }
+    ```
+
+* **Success Response**
+
+  * **Code** `201`
+
+* **Error Response**
+
+  * **Code** `401 UNAUTHORIZED`
+
+  OR
+
+  * **Code** `400 Bad Request`  
+    **Content**
+    ```
+    {
+        "message": "Provided data did not pass validation",
+        "errors": [
+            "[sellingTo]" => "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
+            "[longDescription][1]" => "The value 345 is not a valid string.",
+        ]
+    }
+    ```
+
+  OR
+
+  * **Code** `404 Not Found`
+
+*Note: Updating variant/main variant is not supported yet*
+
+**Delete Product**
+----
+
+* **URL**
+
+  `/api/v1/products/{uuid}`
+
+* **Method**
+
+  `DELETE`
+
+*  **URL Params**
+
+   **Required**
+
+   `uuid=[string]`
+
+* **Success Response**
+
+  * **Code** `204`
+
+* **Error Response**
+
+  * **Code** `400 Bad Request`  
+    **Content**
+    ```
+    {
+        "code": 400,
+        "message": "This UUID is not valid: f564da76-6d51-4ecd-b004-895c8019a23"
```

Comment:
More obvious invalid ID would be more helpful. 
```suggestion
        "message": "This UUID is not valid: 123"
```

Right now I was thinking if it's malformed UUID or not existing one in a database